src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -290,23 +290,73 @@ void Instance::HandleAddOrUpdateWiFiNetwork(HandlerContext & ctx, const Commands
 {
     MATTER_TRACE_EVENT_SCOPE("HandleAddOrUpdateWiFiNetwork", "NetworkCommissioning");
     Commands::NetworkConfigResponse::Type response;
-    response.networkingStatus = ToClusterObjectEnum(mpDriver.Get<WiFiDriver *>()->AddOrUpdateNetwork(req.ssid, req.credentials));
+    MutableCharSpan debugText;
+#if CHIP_CONFIG_NETWORK_COMMISSIONING_DEBUG_TEXT_BUFFER
+    char debugTextBuffer[CHIP_CONFIG_NETWORK_COMMISSIONING_DEBUG_TEXT_BUFFER];
+    debugText = MutableCharSpan(debugTextBuffer);
+#endif
+    uint8_t outNetworkIndex   = 0;
+    response.networkingStatus = ToClusterObjectEnum(
+        mpDriver.Get<WiFiDriver *>()->AddOrUpdateNetwork(req.ssid, req.credentials, debugText, &outNetworkIndex));
+
+    size_t debugTextLen = strnlen(debugText.data(), debugText.size());
+    if (debugTextLen > 0)
+    {
+        response.debugText.SetValue(CharSpan(debugText.data(), debugText.size()));
+    }
+    if (response.networkingStatus == NetworkCommissioningStatus::kSuccess)
+    {
+        response.networkIndex.SetValue(outNetworkIndex);
+    }
     ctx.mCommandHandler.AddResponseData(ctx.mRequestPath, response);
 }
 
 void Instance::HandleAddOrUpdateThreadNetwork(HandlerContext & ctx, const Commands::AddOrUpdateThreadNetwork::DecodableType & req)
 {
     MATTER_TRACE_EVENT_SCOPE("HandleAddOrUpdateThreadNetwork", "NetworkCommissioning");
     Commands::NetworkConfigResponse::Type response;
-    response.networkingStatus = ToClusterObjectEnum(mpDriver.Get<ThreadDriver *>()->AddOrUpdateNetwork(req.operationalDataset));
+    MutableCharSpan debugText;
+#if CHIP_CONFIG_NETWORK_COMMISSIONING_DEBUG_TEXT_BUFFER
+    char debugTextBuffer[CHIP_CONFIG_NETWORK_COMMISSIONING_DEBUG_TEXT_BUFFER];
+    debugText = MutableCharSpan(debugTextBuffer);
+#endif
+    uint8_t outNetworkIndex   = 0;
+    response.networkingStatus = ToClusterObjectEnum(
+        mpDriver.Get<ThreadDriver *>()->AddOrUpdateNetwork(req.operationalDataset, debugText, &outNetworkIndex));
+
+    size_t debugTextLen = strnlen(debugText.data(), debugText.size());
+    if (debugTextLen > 0)
+    {
+        response.debugText.SetValue(CharSpan(debugText.data(), debugText.size()));
+    }
+    if (response.networkingStatus == NetworkCommissioningStatus::kSuccess)
+    {
+        response.networkIndex.SetValue(outNetworkIndex);
+    }
     ctx.mCommandHandler.AddResponseData(ctx.mRequestPath, response);
```

Comment:
Instead of having 4 copies of this code, could we have a helper function for it, please?